components/kserve/kserve.go

```diff
@@ -103,34 +103,33 @@ func (k *Kserve) ReconcileComponent(ctx context.Context, cli client.Client, owne
 		"odh-model-controller": "RELATED_IMAGE_ODH_MODEL_CONTROLLER_IMAGE",
 	}
 
-	enabled := k.GetManagementState() == operatorv1.Managed
+	enabled := k.GetManagementState() == operatorv1.Managed && dscispec.ServiceMesh != nil
 	monitoringEnabled := dscispec.Monitoring.ManagementState == operatorv1.Managed
 	platform, err := deploy.GetPlatform(cli)
 	if err != nil {
 		return err
 	}
 
 	if !enabled {
-		if err := k.removeServerlessFeatures(dscispec); err != nil {
-			return err
-		}
-	} else {
-		// Configure dependencies
-		if err := k.configureServerless(cli, dscispec); err != nil {
+		err := k.removeServerlessFeatures(dscispec)
+		// should return here, why continue?
+		return err
```

Comment:
i added this comments , just to clarify 
in the old code:
if we disable kserve, we run `removeServerlessFeatures` function and only when it is err, we return 
then continue with `deploy.DeployManifestsFromPath` etc
so my question is:
if we disable kserve, we run `removeServerlessFeatures` function , should it return as  well when there is no err, right?

Reply:
after think twice,  i understand why we did this in the past.
we do need to continue with deploy.DeployManifestsFromPath() and so on, that is in order to pass the "false" to remove deployed manifests from kserve.  
i will revert this part

Reply:
ping @israel-hdez here for another round of eyes

Reply:
> we do need to continue with deploy.DeployManifestsFromPath() and so on

Yes, this was it, IIRC. Otherwise, my understanding is that some clean-up would not be done.

Reply:
thanks for confirming

Reply:
> we do need to continue with deploy.DeployManifestsFromPath() and so on, that is in order to pass the "false" to remove deployed manifests from kserve.

this is why I added a note in my previous comment that it might be good to simplify the flow here :)

Reply:
Agreed, 
this part need some refactor, even me forgot the original usage of this "enable" flag :rage4:  
shame on my memory

Reply:
create https://issues.redhat.com/browse/RHOAIENG-4865 to follow up